content/api/commands/intercept.md

```diff
@@ -612,6 +523,220 @@ Clone the <Icon name="github"></Icon> [Real World App (RWA)](https://github.com/
 
 </Alert>
 
+### Request/Response Modification with `routeHandler`
+
+Specifying a `routeHandler` as the last argument to `cy.intercept` is what's required to leverage capabilities beyond just spying routes.
+
+<!-- TODO emphasize the usage of StaticResponse as the routeHandler -->
+
+If a function is passed as the `routeHandler`, it will be called with the intercepted HTTP request:
+
+```js
+cy.intercept('/api', (req) => {
+  // do something with the intercepted request
+})
+```
+
+From here, you can do several things with the intercepted request:
+
+- modify and make assertions on the request like its body, headers, URL, method, etc.
+- stub out the response without interacting with a real back-end
+- pass the request through to its destination and modify or make assertions on the real response on its way back
+- attach listeners to various events on the request
+
+#### Asserting on a request
+
+```js
+cy.intercept('POST', '/users', (req) => {
+  expect(req.body).to.include('Peter Pan')
+})
+```
+
+You can use the request handler callback to make an assertion on the Intercepted Request Object modify the request before it is sent.
+
+#### Controlling the outgoing request
+
+```js
+// modify the request body before it's sent to its destination
+cy.intercept('POST', '/users', (req) => {
+  req.body = {
+    name: 'Peter Pan',
+  }
+})
+
+// add a header to an outgoing request
+cy.intercept('POST', '/users', (req) => {
+  req.headers['x-custom-header'] = 'added by cy.intercept'
+})
+
+// modify an existing header
+cy.intercept('POST', '/users', (req) => {
+  req.headers['Authorization'] = 'Basic YWxhZGRpbjpvcGVuc2VzYW1l'
+})
+```
+
+#### Verifying the request modification
+
+```js
+cy.intercept('POST', '/users', (req) => {
+  req.headers['x-custom-header'] = 'added by cy.intercept'
+}).as('createUser')
+
+cy.get('button.save').click()
+// you can see the headers in the console output by selecting this line in the command log:
+cy.wait('@createUser')
+  // ...or make an assertion:
+  .its('request.headers')
+  .should('have.property', 'x-custom-header', 'added by cy.intercept')
+```
+
+<Alert type="warning">
+
+The request modification cannot be verified by inspecting the browser's network traffic, since the browser logs network traffic _before_ Cypress can intercept it.
```

Comment:
```suggestion
The request modification cannot be verified by inspecting the browser's network traffic (for example, in Chrome DevTools), since the browser logs network traffic _before_ Cypress can intercept it.
```